daphne/src/roles.rs

```diff
@@ -330,7 +330,10 @@ pub trait DapLeader<S>: DapAuthorizedSender<S> + DapAggregator<S> {
 
         if report.encrypted_input_shares.len() != 2 {
             // TODO spec: Decide if this behavior should be specified.
-            return Err(DapAbort::UnrecognizedMessage);
+            return Err(DapAbort::UnrecognizedMessage {
+                detail: "expected exactly two encrypted input shares".into(),
```

Comment:
Consider including the number of aggregate shares we got.
```suggestion
                detail: format!("expected exactly two encrypted input shares; got {}", report.encrypted_input_shares.len()),
```

Reply:
A rule of thumb about error handling: Errors should not leak anything that's not observable on the wire. In this case the list of encrypted input share are visible to the party who gets this error, so it's not sensitive.